packages/babel-plugin-transform-async-to-generator/test/fixtures/async-to-generator/async-arrow-in-method/output.js

```diff
@@ -2,14 +2,11 @@ let TestClass = {
   name: "John Doe",
   testMethodFailure() {
     var _this = this;
-    return new Promise( /*#__PURE__*/function () {
-      var _ref = babelHelpers.asyncToGenerator(function* (resolve) {
+    return new Promise(function (_x) {
+      return babelHelpers.callAsync(function* (resolve) {
         console.log(_this);
         setTimeout(resolve, 1000);
-      });
-      return function (_x) {
-        return _ref.apply(this, arguments);
-      };
-    }());
+      }, this, arguments);
+    });
   }
 };
```

Comment:
If we don't use `OnCallExpression` then we will have to generate output like this.
```
let TestClass = {
    name: "John Doe",
    testMethodFailure() {
      var _this = this;
      return new Promise(function (_x) {
        return (ref =
          ref ||
          babelHelpers.asyncToGenerator(function* (resolve) {
            console.log(_this);
            setTimeout(resolve, 1000);
          })).apply(this, arguments);
      });
    }
};
```